lib/configure.js

```diff
@@ -48,6 +48,7 @@ function configure (gyp, argv, callback) {
         }
       } else {
         log.verbose('`which` succeeded', python, execPath)
+        python = execPath
```

Comment:
I went ahead and filed a PR that adds a regression test: https://github.com/nodejs/node-gyp/pull/756